### PR TITLE
feat: TOOLS-2543 improve asadm help output

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
Examples of different output:

help
```
A tool for interacting with your Aerospike cluster.

Usage:   COMMAND

Commands:

    Default         Print this help message
    asinfo          Provides raw access to the info protocol
    collectinfo     Collects cluster info, system stats, and aerospike conf file for local node
    disable         Exits privileged mode
    enable          Enters privileged mode
    exit            Terminate session
    features        Lists the features in use in a running Aerospike cluster
    info            Provides summary tables for various aspects of Aerospike functionality
    manage          Administrative tasks like managing users, roles, udf, and sindexes
    pager           Turn terminal pager on or off
    show            A collection of commands used to display information about the Aerospike cluster
    summary         Displays summary of Aerospike cluster
    watch           Runs a command for a specified pause and iterations

Run 'help COMMAND' for more information on a command.
```

help show
```
A collection of commands used to display information about the Aerospike cluster.

Usage:  show COMMAND

Commands:

  Default            
  best-practices     Displays any of Aerospike's violated "best-practices"
  config             A collection of commands that display configuration settings
  distribution       A collection of commands that display the distribution of object sizes and time to live
  jobs               A collection of commands that display jobs and associated metadata
  latencies          Displays the server latency histograms
  pmap               Displays partition map analysis of Aerospike cluster
  racks              Displays rack information for a rack-aware cluster
  roles              Displays roles and their assigned privileges, allowlist, and quotas
  roster             Displays roster information per node
  sindex             Displays secondary indexes and static metadata
  statistics         A collection of commands that displays runtime statistics
  stop-writes        Displays all metrics that could trigger stop-writes
  udfs               Displays UDF modules along with metadata
  users              A collection of commands that display user configuration and statistics


Run 'help show COMMAND' for more information on a command.
```

help show config
```
A collection of commands that display configuration settings.

Usage:  show config COMMAND
        or
Usage:  show config [-r] [--flip] [diff] [like <config-substring>]

        -r         - Repeat output table title and row header after
                     every <terminal width> columns.
                     Default: False
        --flip     - Flip output table to show Nodes on Y axis and
                     config on X axis.
        diff       - Only display rows and values that differ between
                     nodes.
        like       - Filter by configuration parameter substring match

Commands:

  Default         Displays security, service, network, and namespace configuration
  namespace       Displays namespace configuration
  network         Displays network configuration
  security        Displays security configuration
  service         Displays service configuration
  xdr             A collection of commands that display xdr configuration


Run 'help show config COMMAND' for more information on a command.
```

help show config xdr
```
A collection of commands that display xdr configuration.

Usage:  show config xdr COMMAND
        or
Usage:  show config xdr [-r] [-flip] [diff] [for <dc-substring>|<ns-substring>] [like <config-substring>]

        -r         - Repeat output table title and row header after
                     every <terminal width> columns.
                     Default: False
        --flip     - Flip output table to show Nodes on Y axis and
                     config on X axis.
        diff       - Only display rows and values that differ between
                     nodes.
        for        - Filter by datacenter or namespace substring match
        like       - Filter by configuration parameter substring match

Commands:

  Default         Displays xdr, xdr datacenter, and xdr namespace configuration
  dc              Displays xdr datacenter configuration
  filter          Displays configured xdr filters
  namespace       Displays xdr namespace configuration


Run 'help show config xdr COMMAND' for more information on a command.
```

help show config xdr dc
```
Displays xdr datacenter configuration.

Usage:  show config xdr dc [-r] [-flip] [diff] [for <dc-substring>] [like <config-substring>]

        -r         - Repeat output table title and row header after
                     every <terminal width> columns.
                     Default: False
        --flip     - Flip output table to show Nodes on Y axis and
                     config on X axis.
        diff       - Only display rows and values that differ between
                     nodes.
        for        - Filter by datacenter substring match
        like       - Filter by configuration parameter substring match
```

### CODE REVIEW

**Comment by hev** [comment]
@jdogmcsteezy can you paste the output of just `asadm help`

**Review by hev — APPROVED** [review]
HUGE improvements!

**Review by dwelch-spike — APPROVED** [review]
LGTM, great improvements!

File: lib/collectinfo_analyzer/health_check_controller.py [hunk 1]
```diff
@@ -14,37 +14,43 @@
...
+        ModifierHelp("-d", "Enable to display extra details of exceptions."),
+        ModifierHelp(
+            "-oc <string>",
+            "Output filter Category. This parameter works if Query file path provided, otherwise health command will work in interactive mode. Format: string of dot (.) separated category levels",
+        ),
+        ModifierHelp(
+            "-wl <string>",
+            "Output filter Warning level. Expected value CRITICAL or WARNING or INFO. This parameter works if Query file path provided, otherwise health command will work in interactive mode.",
+        ),
+    ),
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/health_check_controller.py:46]: It has not been hidden in collectinfo mode. Considering this is mostly used internally I think it makes sense

File: lib/collectinfo_analyzer/show_controller.py [hunk 2]
```diff
@@ -208,10 +258,14 @@ def do_network(self, line):
 
     @CommandHelp(
         "Displays namespace configuration.",
-        "  Options:",
-        "    -r           - Repeat output table title and row header after every <terminal width> columns.",
-        "                   default: False, no repetition.",
-        "    --flip       - Flip output table to show Nodes on Y axis and config on X axis.",
+        usage=f"[-r] [--flip] [{Modifiers.DIFF}] [{Modifiers.FOR} <ns-substring>] [{like_config_usage}]",
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/show_controller.py:261]: They are listed below. Not sure what you mean

File: lib/collectinfo_analyzer/show_controller.py [hunk 3]
```diff
@@ -304,11 +361,22 @@ def do_dc(self, line):
...
+        repeat_modifier_help,
+        flip_config_modifier,
+        diff_row_modifier_help,
+        ModifierHelp(
+            Modifiers.FOR,
+            "Filter by datacenter or namespace substring match",
+        ),
+        like_config_modifier_help,
+    ),
+    usage=f"[-r] [-flip] [{Modifiers.DIFF}] [{Modifiers.FOR} <dc-substring>|<ns-substring>] [{like_config_usage}]",
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/show_controller.py:375]: technically both but I want to only reference --flip from now on

**Pull request merged**